src/test/java/com/solvd/ecommerce/EcommerceTest.java

```diff
@@ -23,18 +28,93 @@ public void setupWebDriver() {
     @Test
     public void verifySearchResultTest() {
         String query = "Maped";
-        webDriver = new ChromeDriver();
+        WebDriver webDriver = new ChromeDriver();
         webDriver.get(ConfigReader.getValue("URL"));
-        HomePage homePage = new HomePage(webDriver);
         ResultPage resultPage = new ResultPage(webDriver);
-        homePage.searchInputEnter(query);
-        homePage.searchButtonClick();
+        Header header = new Header(webDriver);
+        header.searchInputEnter(query);
+        header.searchButtonClick();
         Assert.assertNotEquals(resultPage.getResultListSize(), 0);
         resultPage.printItemData();
+        webDriver.close();
+    }
+
+    @Test
+    public void checkBackgroundChangeTest() {
+        WebDriver webDriver = new ChromeDriver();
+        webDriver.get(ConfigReader.getValue("URL"));
+        HomePage homePage = new HomePage(webDriver);
+        Actions action = new Actions(webDriver);
+        SoftAssert sa = new SoftAssert();
+        List<WebElement> list = homePage.getSidebarMenuList();
+        for (WebElement el : list) {
+            action.moveToElement(el).perform();
+            sa.assertTrue(el.getAttribute("class").contains("main-nav__list__li_wnav_active"));
+        }
+        sa.assertAll();
+        webDriver.close();
+    }
+
+    @Test
+    public void checkPinButtonTest() {
+        SoftAssert sa = new SoftAssert();
+        WebDriver webDriver = new ChromeDriver();
+        webDriver.get(ConfigReader.getValue("URL"));
+        HomePage homePage = new HomePage(webDriver);
+        List<String> classList = homePage.unpinCategoryMenu();
+
+        classList.forEach(el -> sa.assertTrue(el.contains("mpgs-nopin")));
+        sa.assertAll();
+        webDriver.close();
+    }
+
+    @Test
+    public void checkInputResetTest() {
+        String query = "Maped";
+        WebDriver webDriver = new ChromeDriver();
+        webDriver.get(ConfigReader.getValue("URL"));
+        Header header = new Header(webDriver);
+        header.searchInputEnter(query);
+        header.clickResetButton();
+        Assert.assertTrue(header.isSearchInputEmpty());
+        webDriver.close();
+    }
+
+    @Test
+    public void checkSearchFilterResultTest() {
+        String query = "английский";
```

Comment:
code viewed is good from my side, some text that is found during the tests you can move in DataProvider, it will be more flexible

Reply:
Great idea, thanks!